lib/gltfPipeline.js

```diff
@@ -117,41 +116,52 @@ function processFile(inputPath, options, callback) {
     });
 }
 
-function writeFile(gltf, outputPath, options, callback) {
+function writeFile(gltf, outputPath, options) {
     var fileExtension = path.extname(outputPath);
     var binary = defaultValue(options.binary, false);
     var embed = defaultValue(options.embed, true);
     var embedImage = defaultValue(options.embedImage, true);
     var createDirectory = defaultValue(options.createDirectory, true);
-    if (binary || fileExtension === '.glb') {
-        writeBinaryGltf(gltf, {
-            outputPath : outputPath,
-            embed : embed,
-            embedImage : embedImage,
-            createDirectory : createDirectory
-        }, callback);
-    } else {
-        writeGltf(gltf, outputPath, embed, embedImage, createDirectory, callback);
-    }
+
+    return new Promise(function() {
```

Comment:
If you are making a new promise, your function should at least take a `resolve` parameter, which you will call when the function completes. If an exception can be thrown, you should take a `reject` parameter as well and `reject(err)`. That being said, you shouldn't need to write a new promise very often. For example, in this function instead of making a new promise, you should just return `writeBinaryGltf` or `writeGltf` since they are already promises.